server/src/controllers/parcels/parcelDetails.ts

```diff
@@ -2,8 +2,9 @@ import parcelDetailsDB from '../../database/queries/parcelsQueries';
 
 const parcelDetails = async (req: any, res: any) => {
   const { id } = req.params;
-  const { rows } = await parcelDetailsDB(id);
-  if (rows.length === 0) {
+  const { userId } = req;
+  const { rowCount, rows } = await parcelDetailsDB(id, userId);
+  if (rowCount.length === 0) {
```

Comment:
rowCount it is a number so it will return 0 or 1 etc, so why you destructed it rowCount.length? 
check your code what return by console.log before push it.